### PR TITLE
GDALMDArray::GetResampled(): take into account good_wavelengths array…

### PR DESCRIPTION
**Pull request by rouault**
… for EMIT dataset orthorectification

When the /sensor_band_parameters/good_wavelengths variable is present, use it to decide of the current array along the band dimension are valid.

### CODE REVIEW

File: gcore/gdalmultidim_gltorthorectification.cpp [hunk 1]
```diff
@@ -124,6 +127,32 @@ class GLTOrthoRectifiedArray final : public GDALPamMDArray
         newAr->m_poSRS.reset(oSRS.Clone());
         newAr->m_poSRS->SetDataAxisToSRSAxisMapping(
             {/*latIdx = */ 1, /* lonIdx = */ 2});
+        if (CPLTestBool(CSLFetchNameValueDef(papszOptions,
+                                             "USE_BAD_WAVELENGTHS", "YES")) &&
```
Comment (elpaso) [inline gcore/gdalmultidim_gltorthorectification.cpp:131]: Is this spelled correctly? Shouldn't it be 
```suggestion
                                             "USE_GOOD_WAVELENGTHS", "YES")) &&
```
according to https://github.com/OSGeo/gdal/pull/10347/files#diff-47bb821c4b705cacab4b4ad82536126da726d7e5c7dcd9ea04e64c7207131e81R8451

File: gcore/gdalmultidim_gltorthorectification.cpp [hunk 2]
```diff
@@ -124,6 +127,32 @@ class GLTOrthoRectifiedArray final : public GDALPamMDArray
...
+            poParent->GetDimensionCount() == 3)
+        {
+            auto poGoodWaveLengths = poRootGroup->OpenMDArrayFromFullname(
```
Comment (elpaso) [inline gcore/gdalmultidim_gltorthorectification.cpp:134]:
```suggestion
            const auto poGoodWaveLengths = poRootGroup->OpenMDArrayFromFullname(
```

File: gcore/gdalmultidim_gltorthorectification.cpp (hunk 1 above)
Comment (rouault) [reply gcore/gdalmultidim_gltorthorectification.cpp]: fixed

File: gcore/gdalmultidim_gltorthorectification.cpp (hunk 2 above)
Comment (rouault) [reply gcore/gdalmultidim_gltorthorectification.cpp]: fixed

**Pull request merged**